app/src/main/java/com/duckduckgo/app/settings/db/SettingsDataStore.kt

```diff
@@ -53,12 +56,15 @@ interface SettingsDataStore {
     var appNotificationsEnabled: Boolean
     fun isCurrentlySelected(clearWhatOption: ClearWhatOption): Boolean
     fun isCurrentlySelected(clearWhenOption: ClearWhenOption): Boolean
+    fun isCurrentlySelected(fireAnimation: FireAnimation): Boolean
     fun hasBackgroundTimestampRecorded(): Boolean
     fun clearAppBackgroundTimestamp()
 }
 
 class SettingsSharedPreferences @Inject constructor(private val context: Context) : SettingsDataStore {
```

Comment:
@cmonfortep now that you're touching this class maybe is a good moment to clean the DI. SettingsSharedPreferences is both annotated with `@Inject` for constructor injection and also has a provides method in `BrowserAutoCompleteModule`. I think we should remove one of them

Reply:
I can do it. I will take a look tomorrow.

Reply:
I think the `BrowserAutoCompleteModule` should be deleted and then add `@Binds abstract fun bindSettingsDataStore(settingsDataStore: SettingsSharedPreferences): SettingsDataStore` into the `StoreModule`

Reply:
If `SettingsSharedPreferences` is an implementation detail of `SettingsDataStore`, I think it is best not having a `Binds`. To avoid getting the `SettingsSharedPreferences` type (which should be private).
I think in the `StoreModule` we could add a provider method that takes `Context` and returns `SettingsDataStore`

Reply:
Haven't checked but I assumed we were currently returning `SettingsSharedPreferences` which was an injected parameter in the `BrowserAutoCompleteModule` @Provides method hence why `@Binds` made sense to me. If not, then disregard :)

Reply:
By the way, isn't it a very common pattern having an implementation constructor using `@Inject` and then using `@Binds` to provide it? I'm sure I've seen that multiple times pretty much everywhere. Like you say, you could then inject the actual implementation which you should not be able to so... I think we should get rid of the whole `StoreModule` to sort that out. Maybe it makes more sense to create a LHF task to address all those issues rather than just fixing one here?

Reply:
Not sure if it is a common pattern, I'd say it is an anti-pattern. That basically makes both types available through DI, which may not be desired (I think it should not be desired). 
The purpose of having an interface is to hide the type that implements it. Exposing both the implementing type and interface type partly defeats the purpose of having an interface in the first place, but also may be dangerous. It is very easy to miss that I should be asking for one not the other, not clear which one I should be asking for, if one is marked as `Singleton` (or differently scoped) it can even cause bugs. 

Imo a better pattern would be to even have the implementing type, in this case the `SettingsSharedPreferences` as private so that it is not visible to anybody. But if that can't be done, then use a provider that only exposes the final `interface` type.